### PR TITLE
feat(implement-new-functionality-links): Label in name custom rule

### PR DESCRIPTION
**Pull request by madalynrose**
#### Details

This PR implements the custom rule to be run for the new "Label in name" requirement for the Links assessment. 
It includes:
* a new `LabelInNameUtils` file that includes:
  * a message about the code being based on code in axe-core with links to axe-core and the MPL license
  * `getLabelInNameMatches`: determines whether a node should be included in the rule
  * `getLabelInNameData`: returns instance data for the node (`accessibleName`, `visibleText`, `url`, and `labelContainsVisibleText`) 
  * `getVirtualNode`: a utility function that allows us to test whether visible text is calculated properly
* `LabelInName` custom rule:
 * `evaluateLabelInName` sets instance data to the value returned by `getLabelInNameData`, always returns true to surface all instances
 
##### Motivation

feature work

##### Context

This work will be used by the new test created in #5961.

#### Pull request checklist

- [N/A] Addresses an existing issue: #0000
- [x] Ran `yarn null:autoadd`
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/assessments/links/test-steps/link-purpose.tsx [hunk 1]
```diff
@@ -77,11 +77,6 @@ const propertyBagConfig: PropertyBagColumnRendererConfig<LinkPurposePropertyBag>
         displayName: 'Accessible name',
         defaultValue: NoValue,
     },
-    {
-        propertyName: 'accessibleDescription',
```
Comment (shanisebarona) [inline src/assessments/links/test-steps/link-purpose.tsx:81]: This is just a curiosity question, how come this gets removed from link-purpose?

**Review by shanisebarona — APPROVED** [review]
LGTM 🚀

File: src/assessments/links/test-steps/link-purpose.tsx (hunk 1 above)
Comment (madalynrose) [reply src/assessments/links/test-steps/link-purpose.tsx:81]: ah this is a mistake! reverting this! thanks!

File: src/scanner/label-in-name-utils.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/*
```
Comment (madalynrose) [inline src/scanner/label-in-name-utils.ts:4]: @DaveTryon do you think this message is sufficient to satisfy the copy-left license agreement?

Comment (madalynrose) [reply src/scanner/label-in-name-utils.ts]: After discussion with CELA, it is noted that these changes don't rise to the level of copy-lefting and attribution is covered in our NOTICE file.

**Pull request merged**